app/src/main/java/org/fossasia/susi/ai/activities/MainActivity.java

```diff
@@ -741,46 +734,58 @@ public void execute(Realm bgRealm) {
                                         rvChatFeed.getRecycledViewPool().clear();
                                         recyclerAdapter.notifyItemChanged((int) id);
                                         final String setMessage = answer;
-
+                                        final int counterValue = counter;
+                                        actionType = response.body().getAnswers().get(0).getActions().get(counter).getType();
                                         final Handler delayHandler = new Handler();
                                         delayHandler.postDelayed(new Runnable() {
                                             @Override
                                             public void run() {
-                                                addNewMessage(setMessage, ismap, isHavingLink, isPieChart, isSearchResult, datumList);
+                                                actionType = susiResponse.getAnswers().get(0).getActions().get(counterValue).getType();
```

Comment:
@chiragw15 After #412 got merged, that cause was produced. So I thought this may fix all the above. Let me check where the cause was produced. 
Thanks.